src/v/v8_engine/executor.cc

```diff
@@ -152,8 +153,55 @@ ss::future<>
 executor_service::start(ss::alien::instance& instance, int64_t size) {
     co_await _executor.start_single(
       std::ref(instance), _core_for_executor_thread, size);
+    co_await _code_database.start_single();
 }
 
-ss::future<> executor_service::stop() { return _executor.stop(); }
+ss::future<> executor_service::stop() {
+    co_await _code_database.stop();
+    co_await _executor.stop();
+}
+
+ss::future<>
+executor_service::insert_or_assign(coproc::script_id id, iobuf code) {
```

Comment:
Comments from prev review:
Should use foreign_ptr here?

Reply:
if i undersand this correctly caller will use that interface in this way: 

```c++
executor_service.local().insert_or_assign(id, buffer)
```
In this case the `buffer` doesn't have to be a foreign pointer.

Reply:
> if i undersand this correctly caller will use that interface in this way:
> 
> ```c++
> executor_service.local().insert_or_assign(id, buffer)
> ```
> 
> In this case the `buffer` doesn't have to be a foreign pointer.

It was in prev version. In this executor_service is single_core_service.
Inside `insert_or_assign` it calls `invoke_on` for code_database